test/functional/testUtils.js

```diff
@@ -0,0 +1,230 @@
+            } else if (key === 'static_attributes') {
+                type.staticAttributes = group.static_attributes;
+            } else if (key !== 'resource') {
+                type[key] = group[key];
+            }
+        }
+    }
+    type.service = service;
+    type.subservice = subservice;
+    return { name: group.entity_type, type: type };
+}
+
+/**
+ * Test Case function
+ * @brief Sends a measure to the IoT Agent and validates the response
+ * and validates the Context Broker expectation
+ *
+ * @param {Object} measure      Measure to be sent to the IoT Agent
+ * @param {Object} expectation  Expectation for the Context Broker
+ * @param {Object} env          Environment variables
+ * @param {Object} config       IoTA Configuration object
+ * @param {String} type         Type of test (multientity or multimeasure)
+ * @param {String} transport    Transport to be used (Lib, HTTP or MQTT). If not specified, Lib is used.
+ * @param {Boolean} regex       If true, the expectation is treated as a regex
+ */
+async function testCase(measure, expectation, provision, env, config, type, transport, regex) {
+    let receivedContext = [];
+    let cbMockRoute = '';
+    // Set the correct route depending if the test is multientity or not
+    if (type === 'multientity') {
+        cbMockRoute = '/v2/op/update';
+    } else {
+        cbMockRoute = '/v2/entities?options=upsert';
+    }
+
+    // Set the correct mock times depending if the test is multimeasure or not
+    // based on the length of the expectation array
+    let mockTimes = 1;
+    if (expectation.length > 1) {
+        mockTimes = expectation.length;
+    }
+
+    let contextBrokerMock = nock('http://192.168.1.1:1026')
+        .matchHeader('fiware-service', env.service)
+        .matchHeader('fiware-servicepath', env.servicePath)
+        .post(cbMockRoute, function (body) {
+            mockTimes === 1 ? (receivedContext = body) : receivedContext.push(body); // Save the received body for later comparison
+            return true;
+        })
+        .times(mockTimes)
+        .reply(204);
+
+    // Send a measure to the IoT Agent and wait for the response
+    if (transport === 'MQTT') {
+        try {
+            let client = await MQTT.connectAsync('mqtt://' + config.mqtt.host);
+            await client.publish('/' + measure.qs.k + '/' + measure.qs.i + '/attrs', JSON.stringify(measure.json));
+            await client.end();
+        } catch (error) {
+            expect.fail(ERR_MQTT + error);
+        }
+    } else if (transport === 'HTTP') {
+        // HTTP
+        const response = await sendMeasureHttp(measure);
+        // Validate the response status code and the response body
+        expect(response.statusCode, ERR_MEAS_CODE).to.equal(200);
+        expect(response.body, ERR_MEAS_BODY).to.be.empty;
+    } else {
+        const response = await sendMeasureIotaLib(measure, provision);
+        // expect(response, ERR_MEAS_BODY).to.be.empty;
```

Comment:
Lefover comment? Or something pending to fix?

Reply:
Solved here: https://github.com/telefonicaid/iotagent-node-lib/pull/1525/commits/05b86a5e4bfecc4cce4f320b810334a4b590fc64